colorama/ansi.py

```diff
@@ -100,3 +101,33 @@ class AnsiStyle(AnsiCodes):
 Back   = AnsiBack()
 Style  = AnsiStyle()
 Cursor = AnsiCursor()
+
+fore_values = set(Fore.__dict__.values())
+back_values = set(Back.__dict__.values())
+style_values = set(AnsiStyle.__dict__.values())
+
+fore_stack = [Fore.RESET]
+back_stack = [Back.RESET]
+style_stack = [Style.NORMAL]
+
+@contextlib.contextmanager
+def colorama(*args):
+    for arg in args:
+        print(arg, end='');
+        if arg in fore_values:
+            fore_stack.append(arg)
+        elif arg in back_values:
+            back_stack.append(arg)
+        elif arg in style_values:
+            style_stack.append(arg)
+    yield
+    for arg in reversed(args):
```

Comment:
I think we don't need this loop. If we pass more than one `Fore` value, for example, we are going to print the colors in reverse (except the first one), but we don't need to print any color codes after the text has been written. The reason this code works is because the stacks always end with a reset - so all we need to do after the `yield` is print the reset codes, or only print `Style.RESET_ALL`.